kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/submatching/StringPartialMatches.kt

```diff
@@ -2,16 +2,25 @@ package io.kotest.submatching
 
 import io.kotest.assertions.AssertionsConfig
 
-internal fun describePartialMatchesInStringForSlice(expectedSlice: String, value: String) =
+class StringPartialMatch(val expected: String, val value: String) {
```

Comment:
I don't understand why we need all these guys as part of the public API.
Where is access to them being public required?